Makefile

```diff
@@ -1,14 +1,75 @@
 #!make
 
+ifneq ($(OS),Windows_NT)
+POSIXSHELL := 1
+KC_HOST_ENTRY := $(shell grep "keycloak" /etc/hosts)
+else
+POSIXSHELL :=
+KC_HOST_ENTRY := $(shell findstr "keycloak" C:\Windows\System32\drivers\etc\hosts)
+endif
+
+local-dev: one-time-local-dev-env-setup
+restore-dev: restore-last-env
+rebuild-all-local: reset | project pause-30 create-local-keycloak-users generate-rand1000 rebuild-all-local-friendly-message
 backend: backend-build | backend-run
 database: database-build | database-run
 frontend: frontend-build | frontend-run
 project: project-build | project-run
 rebuild: project-build
-reset:	stop | clean
+reset:  stop | clean
 database-seed: database-dump | database-dump-seed
 database-seed-local: database-dump | database-dump-seed-local
 
+one-time-local-dev-env-setup:
+	@echo "+\n++ Setting up your local development environment"
+	@echo "++ with local authentication and db.  Run this once only."
+	@echo "++ Your last configuration files were saved with the ending '*-last-backup'."
+ifneq ($(POSIXSHELL),)
+ifeq ($(KC_HOST_ENTRY),)
+	@echo "++ Adding required keycloak entry to hosts file:"
+	@echo "127.0.0.1       localhost       keycloak" | sudo tee -a /etc/hosts;
```

Comment:
see MDS-1221

Reply:
from what I remember yes you can run keycloak locally without updating the hosts file.

But when you get redirected from your frontend on the browser, it needs to say `keycloak:8080` instead of `localhost:8080` because of CORS issues. Now you browser doesnt know `keycloak` is. So we add the host entry to loopback